source/Octopus.Tentacle.Tests.Integration/ClientScriptExecutionCanBeCancelledWhenRetriesAreDisabled.cs

```diff
@@ -185,42 +160,34 @@ public async Task DuringStartScript_ScriptExecutionCanBeCancelled(TentacleConfig
-                case ExpectedFlow.CancelRpcAndExitImmediately:
-                    latestException.Should().BeTaskOrOperationCancelledException().And.NotBeOfType<OperationAbandonedException>();
-                    break;
-                case ExpectedFlow.AbandonRpcThenCancelScriptThenCompleteScript:
-                    latestException?.Should().BeOfType<HalibutClientException>();
-                    break;
-                default:
-                    throw new NotSupportedException();
-            }
+                // Should have cancelled the RPC call and exited immediately
+                actualException.Should().BeRequestCancelledException(rpcCallStage);
 
-            // If the rpc call could not be cancelled it cancelled fairly quickly e.g. we are not waiting for an rpc call to timeout
-            cancellationDuration.Should().BeLessOrEqualTo(TimeSpan.FromSeconds(30));
+                // We should have cancelled the RPC call quickly and existed
+                cancellationDuration.Should().BeLessOrEqualTo(TimeSpan.FromSeconds(15));
 
-            // The expected RPC calls were made - this also verifies the integrity of the test
-            recordedUsages.For(nameof(IAsyncClientScriptServiceV2.StartScriptAsync)).Started.Should().Be(1);
-            recordedUsages.For(nameof(IAsyncClientScriptServiceV2.GetStatusAsync)).Started.Should().Be(0, "Test should not have not proceeded past StartScript before being Cancelled");
-
-            switch (expectedFlow)
+                recordedUsages.ForStartScriptAsync().Started.Should().Be(1);
+                recordedUsages.ForGetStatusAsync().Started.Should().Be(0, "Test should not have not proceeded past StartScript before being Cancelled");
+                recordedUsages.ForCancelScriptAsync().Started.Should().Be(0);
+                recordedUsages.ForCompleteScriptAsync().Started.Should().Be(0);
+            }
+            else // Transferring
             {
-                case ExpectedFlow.CancelRpcAndExitImmediately:
-                    recordedUsages.For(nameof(IAsyncClientScriptServiceV2.CancelScriptAsync)).Started.Should().Be(0);
-                    recordedUsages.For(nameof(IAsyncClientScriptServiceV2.CompleteScriptAsync)).Started.Should().Be(0);
-                    break;
-                case ExpectedFlow.AbandonRpcThenCancelScriptThenCompleteScript:
-                    recordedUsages.For(nameof(IAsyncClientScriptServiceV2.CancelScriptAsync)).Started.Should().BeGreaterOrEqualTo(1);
-                    recordedUsages.For(nameof(IAsyncClientScriptServiceV2.CompleteScriptAsync)).Started.Should().Be(1);
-                    break;
-                default:
-                    throw new NotSupportedException();
+                // Assert the CancelScript and CompleteScript flow happened fairly quickly
+                actualException.Should().BeScriptExecutionCancelledException();
+
+                // Assert the CancelScript and CompleteScript flow happened fairly quickly
```

Comment:
This feels like a repeat of the comment two lines above.

Actually, the comments in the `Connecting` state are kind of repeated as well (but less so)

Reply:
Done